internal/cmd/base/servers.go

```diff
@@ -548,20 +549,37 @@ func (b *Server) RunShutdownFuncs() error {
 }
 
 func (b *Server) ConnectToDatabase(dialect string) error {
-	dbase, err := gorm.Open(dialect, b.DatabaseUrl)
+	dbType, err := db.StringToDbType(dialect)
+	if err != nil {
+		return fmt.Errorf("unable to create db object with dialect %s: %w", dialect, err)
+	}
+	dbase, err := db.Open(dbType, b.DatabaseUrl)
 	if err != nil {
 		return fmt.Errorf("unable to create db object with dialect %s: %w", dialect, err)
 	}
 	if b.DatabaseMaxOpenConnections == 1 {
 		return fmt.Errorf("unable to create db object with dialect %s: %s", dialect, "max_open_connections must be unlimited by setting 0 or at least 2")
 	} else {
-		dbase.DB().SetMaxOpenConns(b.DatabaseMaxOpenConnections)
+		underlyingDB, err := dbase.DB()
+		if err != nil {
+			return fmt.Errorf("unable retreive db: %w", err)
+		}
+		underlyingDB.SetMaxOpenConns(b.DatabaseMaxOpenConnections)
 	}
 	b.Database = dbase
 	if os.Getenv("BOUNDARY_DISABLE_GORM_FORMATTER") == "" {
-		gorm.LogFormatter = db.GetGormLogFormatter(b.Logger)
-		b.Database.SetLogger(db.GetGormLogger(b.Logger))
+		newLogger := logger.New(
+			db.GetGormLogger(b.Logger),
+			logger.Config{
+				LogLevel: logger.Error, // Log level
+				Colorful: false,        // Disable color
+			},
+		)
+		dbase = dbase.Session(&gorm.Session{Logger: newLogger})
```

Comment:
Likewise I think this could be refactored into the db package so that this package would not need to import `gorm` at all.

Reply:
this is can be addressed in #1535